modules/flight_interface/flight_interface_worker.py

```diff
@@ -45,4 +49,8 @@ def flight_interface_worker(
         if not result:
             continue
 
+        with odometry_mutex:
+            global current_odometry
+            current_odometry = value
```

Comment:
This can be replaced with a queue of size 1. Whenever new odometry is available, attempt to remove the item (catching the exception if the queue is empty) and then push the new odometry.